From 828856441b557017ebe9e718c7c5bfbe8ff70d6e Mon Sep 17 00:00:00 2001
From: Roger Leigh <rleigh@debian.org>
Date: Tue, 5 Apr 2011 14:03:40 +0100
Subject: [PATCH] Add FHS exception for /run

Document the replacement of /var/run with /run and /var/lock
with /run/lock.  Also document the requirements for init scripts.
---
 policy.sgml |   39 +++++++++++++++++++++++++++------------
 1 files changed, 27 insertions(+), 12 deletions(-)

diff --git a/policy.sgml b/policy.sgml
index ec605c6..4d3602b 100644
--- a/policy.sgml
+++ b/policy.sgml
@@ -6205,10 +6205,21 @@ install -m644 debian/shlibs.<var>package</var> debian/<var>package</var>/DEBIAN/
               <item>
                 <p>
                   The following directories in the root filesystem are
-                  additionally allowed: <file>/sys</file> and
-                  <file>/selinux</file>. <footnote>These directories
-                  are used as mount points to mount virtual filesystems
-                  to get access to kernel information.</footnote>
+                  additionally allowed: <file>/run</file>,
+		  <file>/sys</file> and <file>/selinux</file>.
+		  <footnote>The <file>/run</file> directory is a
+                  replacement for <file>/var/run</file>, and its
+                  subdirectory <file>/run/lock</file> is a replacement for
+                  <file>/var/lock</file>.  These changes have been
+                  adopted by most distributions and have been proposed
+                  for inclusion in a future revision of the FHS.  Both
+                  are expected to be temporary filesystems, whose
+                  purpose is storage of ephemeral system state which
+                  should not be preserved across reboot.
+                  The <file>/sys</file> and <file>/selinux</file>
+                  directories are used as mount points to mount
+                  virtual filesystems to get access to kernel
+                  information.</footnote>
                 </p>
               </item>
 	      <item>
@@ -6719,14 +6730,18 @@ test -f <var>program-executed-later-in-script</var> || exit 0
 	  </p>
 
 	  <p>
-	    <file>/var/run</file> and <file>/var/lock</file> may be mounted
-	    as temporary filesystems<footnote>
-		For example, using the <tt>RAMRUN</tt> and <tt>RAMLOCK</tt>
-		options in <file>/etc/default/rcS</file>.
-	    </footnote>, so the <file>init.d</file> scripts must handle this
-	    correctly. This will typically amount to creating any required
-	    subdirectories dynamically when the <file>init.d</file> script
-	    is run, rather than including them in the package and relying on
+	    <file>/var/run</file> and <file>/var/lock</file> should be
+	    symlinks to <file>/run</file> and <file>/run/lock</file>,
+	    respectively, which are temporary filesystems whose
+	    contents are not preserved across reboot.  This
+	    arrangement may also be satisfied through equivalent
+	    means, for example bind or nullfs mounts.  Because the
+	    presence of files or directories in any of these
+	    directories is not guaranteed, <file>init.d</file> scripts
+	    must handle this correctly. This will typically amount to
+	    creating any required subdirectories dynamically when
+	    the <file>init.d</file> script is run, rather than
+	    including them in the package and relying on
 	    <prgn>dpkg</prgn> to create them.
 	  </p>
 	</sect1>
-- 
1.7.4.1

